### DISCUSSION TITLE
ComboBox.show_ui().response.changed() not firing

### DISCUSSION THREAD

**Discussion by vbeffa** [emilk/egui#923] [Q&A] [answered]
I'm trying to add a listener for a combobox when its value changes. It seems `ComboBox.show_ui().response.changed()` should be what to use, but it's not working. `clicked()` fires but only when the combobox is opened, not when a new value is selected, so the updated value is not seen by `clicked()` until the next time it's opened. Is there another function I should be using?

**Comment by vbeffa** [comment, marked as answer]
It looks like you need to add `if xxx.clicked()` to each value inside the combobox:

```rust
egui::ComboBox::from_label( "Select one!")
    .selected_text(format!("{:?}", selected))
    .show_ui(ui, |ui| {
        if ui.selectable_value(&mut selected, Enum::First, "First").clicked() {
          // handle click
        };
        if ui.selectable_value(&mut selected, Enum::Second, "Second").clicked() {
          // handle click
        };
        if ui.selectable_value(&mut selected, Enum::Third, "Third").clicked() {
          // handle click
        };
    }
);
```

**Reply by emilk** [reply]
You can also store the before-value of the enum then compare after showing the combo-box. Less code :)
